nftest/common.py

```diff
@@ -53,39 +53,30 @@ def print_version_and_exist():
     print(__version__, file=sys.stdout)
     sys.exit()
 
+
 def setup_loggers():
     """ Initialize loggers for both init and run """
-    _ = generate_logger('NFTest')
-    _ = generate_logger('NFTestInit')
+    # Always log times in UTC
+    logging.Formatter.converter = time.gmtime
 
-# pylint: disable=W0212
-def generate_logger(logger_name:str):
-    """ Generate program-specific logger """
     _env = NFTestENV()
-    try:
-        log_level = logging._checkLevel(_env.NFT_LOG_LEVEL)
-    except ValueError:
-        log_level = logging._checkLevel('INFO')
-
-    logger = logging.getLogger(logger_name)
-    logger.setLevel(log_level)
-
-    formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
-    logging.Formatter.converter = time.gmtime
 
+    # Make a file handler that accepts all logs
     try:
         file_handler = logging.FileHandler(_env.NFT_LOG)
+        file_handler.setLevel(logging.DEBUG)
```

Comment:
I'm not opposed to it but just wondering, is it usually common practice to set the log level for log files to the lowest level to capture all messages?

Reply:
It's a philosophical debate.

My take is that a user is limited in how they can interact with the console logs, so those should be relatively terse and easy to scan. In handwavy terms there's a quadratic/polynomial/something growth in the number of log statements per level; for every 1 `FATAL` you get 3 `ERROR`s, 10 `WARNINGS`, 50 `INFO`s, 200 `DEBUG`s, etc. Your important logs can easily get drowned out in a sea of informational messages.

On the other hand, once logs are in a file you've got [so](https://lnav.org/) [many](https://github.com/rcoh/angle-grinder) [different](https://www.datadoghq.com/knowledge-center/log-aggregation/) [tools](https://www.solarwinds.com/log-analyzer/use-cases/log-aggregation-tool) to parse/filter/consolidate them, so you can afford to dump everything in there.

I tend to draw the line of what to save between `DEBUG` and `TRACE` (which python doesn't use) - that level is generally used for logging _every single action_ a program takes and are generally only turned on when you specifically want them.

TL/DR: Disk space is cheap and good log viewers exist, so I'd say we should err on the side of saving everything.

Reply:
That makes sense!